datafusion/core/src/execution/context.rs

```diff
@@ -1833,6 +1860,29 @@ impl From<&SessionState> for TaskContext {
     }
 }
 
+fn create_dialect_from_str(dialect_name: &str) -> Box<dyn Dialect> {
```

Comment:
What would you think about putting this in a PR  upstream in sqlparser-rs?  I can do so if you agree